HashRegistrar.sol

```diff
@@ -200,13 +222,15 @@ contract Registrar {
         } else if (_value > h.highestBid) {
             // new winner
             // cancel the other bid, burn 0.1%
-            Deed previousWinner = Deed(h.deed);
+            Deed previousWinner = h.deed;
             previousWinner.closeDeed(999);
             
             // set new winner
             h.value = h.highestBid;
             h.highestBid = _value;
             h.deed = sealedBids[seal];
+            bid.setOwner(_owner);
```

Comment:
setOwner should still be up top, otherwise the other conditions that close it won't return any ether to them.